packages/mountable/src/dune/utils/fileUtils.spec.ts

```diff
@@ -95,4 +99,113 @@ describe("fileUtils", () => {
       ).rejects.toThrow("API request failed: 400 Bad Request");
     });
   });
+
+  describe("isZipFile", () => {
+    it("should return true for .zip file extension", () => {
+      expect(isZipFile("test.zip")).toBe(true);
+      expect(isZipFile("TEST.ZIP")).toBe(true);
+      expect(isZipFile("path/to/file.zip")).toBe(true);
+    });
+
+    it("should return true for application/zip MIME type", () => {
+      expect(isZipFile("test", "application/zip")).toBe(true);
+      expect(isZipFile("test.txt", "application/zip")).toBe(true);
+    });
+
+    it("should return true for application/x-zip-compressed MIME type", () => {
+      expect(isZipFile("test", "application/x-zip-compressed")).toBe(true);
+      expect(isZipFile("test.txt", "application/x-zip-compressed")).toBe(true);
+    });
+
+    it("should return false for non-zip files", () => {
+      expect(isZipFile("test.txt")).toBe(false);
+      expect(isZipFile("test.py")).toBe(false);
+      expect(isZipFile("test", "text/plain")).toBe(false);
+      expect(isZipFile("test", "application/json")).toBe(false);
+    });
+
+    it("should return false when no MIME type provided and no .zip extension", () => {
+      expect(isZipFile("test")).toBe(false);
+      expect(isZipFile("test.tar.gz")).toBe(false);
+    });
+  });
+
+  describe("processZipFile", () => {
+    const mockBinaryData = new ArrayBuffer(1024);
+    const mockFileName = "test.zip";
+
+    // Mock Entry type for testing
+    const createMockFileEntry = (
+      filename: string,
+      content: string,
+    ): ZipEntry => {
+      const encodedContent = new TextEncoder().encode(content);
+      return {
+        filename,
+        directory: false,
+        getData: jest.fn().mockResolvedValue(encodedContent),
+      };
+    };
+
+    afterEach(() => {
+      jest.restoreAllMocks();
+    });
+
+    it("should process zip file with multiple text files successfully", async () => {
+      const mockFile1 = createMockFileEntry("file1.txt", "Hello from file1");
+      const mockFile2 = createMockFileEntry(
+        "file2.py",
+        "print('Hello from file2')",
+      );
+      const mockFile3 = createMockFileEntry(
+        "subdir/file3.js",
+        "console.log('Hello from file3')",
+      );
+
+      const mockEntries = [mockFile1, mockFile2, mockFile3];
+
+      const mockGetZipEntries: typeof defaultGetZipEntries = jest
+        .fn()
+        .mockResolvedValue(mockEntries);
```

Comment:
You don't really need a full mock here
```suggestion
      const mockGetZipEntries: typeof defaultGetZipEntries = 
        () => Promise.resolve(mockEntries);
```